formats/gltf/writer.go

```diff
@@ -504,7 +504,10 @@ func (w *Writer) AddTexture(mat PolyformTexture) *TextureInfo {
 	w.images = append(w.images, Image{
 		URI: mat.URI,
 	})
-	w.samplers = append(w.samplers, Sampler{})
+	if mat.Sampler != nil {
+		w.samplers = append(w.samplers, *mat.Sampler)
+	}
+
```

Comment:
I think all textures need a sampler still iirc

```suggestion
	if mat.Sampler != nil {
		w.samplers = append(w.samplers, *mat.Sampler)
	} else {
	        w.samplers = append(w.samplers, Sampler{})
	}

```

Reply:
I have a better idea :) 
```suggestion
	var sampler Sampler{}
	if mat.Sampler != nil {
		sampler = *mat.Sampler
	}
	w.samplers = append(w.samplers, sampler)

```

Reply:
Looks good to me!